src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+    const pagerParent = flatTablePager(page).locator("..");
+    await pagerParent.scrollIntoViewIfNeeded();
+
+    await expect(flatTableHeaderCells(page).first()).not.toBeInViewport();
+    await expect(pagerParent).toBeInViewport();
+
+    for (let i = 0; i < 5; i++) {
+      if (i === 0) {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      }
+    }
+  });
+
+  ([...colorThemes] as [
+    FlatTableProps["colorTheme"],
+    string,
+    string
+  ][]).forEach(([colorTheme, bgColor, brColor]) => {
+    test(`should render in the ${colorTheme} theme`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTableComponent colorTheme={colorTheme} />);
+
+      for (let i = 0; i < 4; i++) {
+        const headerCells = flatTableHeaderCells(page).nth(i);
+        await expect(headerCells).toHaveCSS("background-color", bgColor);
+        await checkCSSOutline(
+          headerCells,
+          "1px",
+          "border-right",
+          "solid",
+          brColor
+        );
+      }
+    });
+  });
+
+  test(`should render with zebra stripes`, async ({ mount, page }) => {
+    await mount(<FlatTableComponent isZebra />);
+
+    for (let i = 0; i < 4; i++) {
+      if (i === 0 || i === 2) {
+        await expect(
+          flatTableBodyRowByPosition(page, i).locator("td").nth(0)
+        ).toHaveCSS("background-color", "rgb(255, 255, 255)");
+      } else {
+        await expect(
+          flatTableBodyRowByPosition(page, i).locator("td").nth(0)
+        ).toHaveCSS("background-color", "rgb(250, 251, 251)");
+      }
+    }
+  });
+
+  ([...sizes] as [FlatTableProps["size"], string, string, number][]).forEach(
+    ([sizeName, padding, fontSize, rowHeight]) => {
+      test(`should render as ${sizeName} size`, async ({ mount, page }) => {
+        await mount(<FlatTableComponent size={sizeName} />);
+
+        await assertCssValueIsApproximately(
+          flatTableHeader(page),
+          "height",
+          rowHeight
+        );
+
+        for (let i = 0; i < 4; i++) {
+          const headerCells = flatTableHeaderCells(page)
+            .nth(i)
+            .locator("div")
+            .nth(0);
+          await expect(headerCells).toHaveCSS("padding-left", padding);
+          await expect(headerCells).toHaveCSS("padding-right", padding);
+          await expect(headerCells).toHaveCSS("font-size", fontSize);
+        }
+      });
+    }
+  );
+
+  heightWidth.forEach((height) => {
+    test(`should render with ${height}px as a height parameter`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTableComponent height={`${height}px`} />);
+
+      await assertCssValueIsApproximately(
+        flatTableWrapper(page),
+        "height",
+        height
+      );
+    });
+  });
+
+  [150, 249, 250, 251, 300].forEach((height) => {
```

Comment:
suggestion: as this array is used in another test later in this file it might be worth storing it as a variable.